### PR TITLE
Issue 1193 navigation components

### LINKED ISSUE #1192: Migrate to use View Binding
https://developer.android.com/topic/libraries/view-binding

### LINKED ISSUE #1193: Migrate to use Navigation component
https://developer.android.com/guide/navigation/navigation-getting-started

### LINKED ISSUE #1274: Improve Splash screen
https://developer.android.com/about/versions/12/features/splash-screen

### LINKED ISSUE #1709: Migrate to use Fragment where possible

### PR DESCRIPTION
**Pull request by DenBond7**
This PR did the migration to Navigation components over the whole app and other important architecture changes

close #1193
close #1274
close #1709
close #1192

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities

### CODE REVIEW

**Comment by DenBond7** [comment]
> When do you think this one could be ready? I'll be writing an issue that I think could be worked on after this one merges.

I'm on the final touches. I'm working on fixing tests. I hope it will be done this week.

File: FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/base/BaseFragment.kt [hunk 1]
```diff
@@ -37,102 +37,103 @@ import com.google.android.material.snackbar.Snackbar
...
+  }
+
+  override fun onDetach() {
+    super.onDetach()
+    LogsUtil.d(loggingTag, "onDetach")
+    lifecycle.removeObserver(connectionLifecycleObserver)
+  }
+
+  open fun onAccountInfoRefreshed(accountEntity: AccountEntity?) {
+
```
Comment (IvanPizhenko) [inline FlowCrypt/src/main/java/com/flowcrypt/email/ui/activity/fragment/base/BaseFragment.kt:136]: is it intentionally empty? If so, please put a comment like `// nothing to do here`

File: FlowCrypt/src/main/res/values-ru/strings.xml [hunk 2]
```diff
@@ -494,4 +493,8 @@
     <string name="error_occurred_during_downloading_att">Произошла ошибка при загрузке.\n\nДетали ошибки:  %1$s</string>
     <string name="no_apps_that_can_handle_intent">Извините, у Вас нет приложений, которые могут открыть этот файл</string>
     <string name="warning_don_not_have_content_app">Просмотр вложения невозможен. Ваша организация требует использовать Content для просмотра вложений, но приложение Content не установлено.\n\nПожалуйста, обратитесь в службу поддержки.</string>
+    <string name="import_keys">Импортировать ключи</string>
+    <string name="error">Ошибка</string>
+    <string name="import_keys_from_different_sources">Вы пытаетесь импортировать ключи с разных источников</string>
```
Comment (IvanPizhenko) [inline FlowCrypt/src/main/res/values-ru/strings.xml:498]: maybe "из разных"?

**Review by IvanPizhenko — CHANGES_REQUESTED** [review]
@DenBond7 some minor comments below.

File: FlowCrypt/src/main/res/values-ru/strings.xml (hunk 2 above)
Comment (DenBond7) [reply FlowCrypt/src/main/res/values-ru/strings.xml:498]: good one. Thanks!

**Review by IvanPizhenko — APPROVED** [review]

**Pull request merged**